frontend-react/src/hooks/UsePagination.test.ts

```diff
@@ -0,0 +1,474 @@
+                2: "11",
+                3: "21",
+                4: "31",
+                5: "41",
+                6: "51",
+                7: "61",
+            },
+            finalPageNum: undefined,
+        });
+    });
+
+    test("with fewer than the requested number of results", () => {
+        const state: PaginationState<SampleRecord> = {
+            currentPageNum: 1,
+            extractCursor,
+            pageSize: 10,
+            pageCursorMap: { 1: "0" },
+            pageResultsMap: {},
+            fetchStartPageNum: 1,
+            fetchNumResults: 61,
+            fetchStartCursor: "0",
+        };
+        const results = createSampleRecords(30);
+        expect(processResultsReducer(state, results)).toStrictEqual({
+            ...state,
+            pageResultsMap: {
+                1: results.slice(0, 10),
+                2: results.slice(10, 20),
+                3: results.slice(20),
+            },
+            pageCursorMap: {
+                1: "0",
+                2: "11",
+                3: "21",
+            },
+            finalPageNum: 3,
+        });
+    });
+});
+
+describe("setCurrentPageReducer", () => {
+    test("with no stored data", () => {
+        const state: PaginationState<SampleRecord> = {
+            currentPageNum: 1,
+            extractCursor,
+            pageSize: 10,
+            pageCursorMap: { 1: "0" },
+            pageResultsMap: {},
+            fetchStartPageNum: 1,
+            fetchNumResults: 61,
+            fetchStartCursor: "0",
+        };
+        expect(setCurrentPageReducer(state, 1)).toStrictEqual({
+            ...state,
+            currentPageNum: 1,
+            fetchNumResults: 61,
+            fetchStartCursor: "0",
+            fetchStartPageNum: 1,
+        });
+    });
+
+    test("when some of the data has been fetched", () => {
+        const state: PaginationState<SampleRecord> = {
+            currentPageNum: 1,
+            extractCursor,
+            pageSize: 10,
+            pageCursorMap: {
+                1: "0",
+                2: "11",
+                3: "21",
+                4: "31",
+                5: "41",
+                6: "51",
+                7: "61",
+            },
+            pageResultsMap: { 1: [], 2: [], 3: [], 4: [], 5: [], 6: [] },
+        };
+        expect(setCurrentPageReducer(state, 6)).toStrictEqual({
+            ...state,
+            currentPageNum: 6,
+            fetchNumResults: 21,
+            fetchStartCursor: "61",
+            fetchStartPageNum: 7,
+        });
+    });
+
+    test("when all of the data has been fetched", () => {
+        const state: PaginationState<SampleRecord> = {
+            currentPageNum: 7,
+            extractCursor,
+            pageSize: 10,
+            pageCursorMap: {
+                1: "0",
+                2: "11",
+                3: "21",
+                4: "31",
+                5: "41",
+                6: "51",
+                7: "61",
```

Comment:
Curious why the cursor map wouldn't update to include pointers to the remaining 3 pages

Reply:
The cursors aren't actually needed for this test case (all of the results have already been fetched, so we don't need to update `fetchStartCursor`), but I filled out the map for completeness.